packages/backend-core/src/cache/docWritethrough.ts

```diff
@@ -0,0 +1,140 @@
+import BaseCache from "./base"
+import { getDocWritethroughClient } from "../redis/init"
+import { AnyDocument, Database, LockName, LockType } from "@budibase/types"
+import * as locks from "../redis/redlockImpl"
+
+import { JobQueue, createQueue } from "../queue"
+import * as context from "../context"
+import * as dbUtils from "../db"
+
+let CACHE: BaseCache | null = null
+async function getCache() {
+  if (!CACHE) {
+    const client = await getDocWritethroughClient()
+    CACHE = new BaseCache(client)
+  }
+  return CACHE
+}
+
+interface ProcessDocMessage {
+  tenantId: string
+  dbName: string
+  docId: string
+  cacheKeyPrefix: string
+}
+
+export const docWritethroughProcessorQueue = createQueue<ProcessDocMessage>(
+  JobQueue.DOC_WRITETHROUGH_QUEUE
+)
+
+docWritethroughProcessorQueue.process(async message => {
+  const { tenantId, cacheKeyPrefix } = message.data
+  await context.doInTenant(tenantId, async () => {
+    const lockResponse = await locks.doWithLock(
+      {
+        type: LockType.TRY_ONCE,
+        name: LockName.PERSIST_WRITETHROUGH,
+        resource: cacheKeyPrefix,
+        ttl: 15000,
```

Comment:
There's no guarantee that the Bull job queue is being processed. If it has an outage and jobs aren't processed for 15 seconds, won't we lose data?

Reply:
The message would retry on that case, but the latest discussed solution does not require it